airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -377,6 +385,67 @@ class FbaShipmentsReports(ReportsAmazonSPStream):
 class VendorInventoryHealthReports(ReportsAmazonSPStream):
     name = "GET_VENDOR_INVENTORY_HEALTH_AND_PLANNING_REPORT"
 
+class BrandAnalyticsSearchTermsReports(ReportsAmazonSPStream):
+    """
+    Field definitions: https://sellercentral.amazon.co.uk/help/hub/reference/G5NXWNY8HUD3VDCW
+    """
+
+    name = "GET_BRAND_ANALYTICS_SEARCH_TERMS_REPORT"
+
+    def parse_document(self, document):
+        parsed = json_lib.loads(document)
+        return parsed.get("dataByDepartmentAndSearchTerm", {})
+
+    def _report_data(
+        self,
+        sync_mode: SyncMode,
+        cursor_field: List[str] = None,
+        stream_slice: Mapping[str, Any] = None,
+        stream_state: Mapping[str, Any] = None,
+    ) -> Mapping[str, Any]:
+         data = super()._report_data(sync_mode, cursor_field, stream_slice, stream_state)
+         options = self.report_options()
+         if options is not None:
+             data.update(self._augmented_data(options))
+
+         return data
+
+    def _augmented_data(self, report_options) -> Mapping[str, Any]:
+        if report_options.get("reportPeriod") is None:
+            return {}
+        else:
+            now = pendulum.now("utc")
+            if report_options["reportPeriod"] == "DAY":
+                now = now.subtract(days=1)
+                data_start_time = now.start_of("day")
+                data_end_time = now.end_of("day")
+            elif report_options["reportPeriod"] == "WEEK":
+                now = now.subtract(weeks=1)
+
+                # According to report api docs
+                # dataStartTime must be a Sunday and dataEndTime must be the following Saturday
+                pendulum.week_starts_at(pendulum.SUNDAY)
+                pendulum.week_ends_at(pendulum.SATURDAY)
+
+                data_start_time = now.start_of("week")
+                data_end_time = now.end_of("week")
+
```

Comment:
can we move this into some util or a static function as this doesn't have any dependency on the class

Reply:
> can we move this into some util or a static function as this doesn't have any dependency on the class

👍🏽  Annotated it as a static but leaving inside the class because the logic is report specific and not reusable